jvm-packages/xgboost4j-spark/docs/index.md

```diff
@@ -0,0 +1,456 @@
+
+```xml
+<repository>
+  <id>GitHub Repo</id>
+  <name>GitHub Repo</name>
+  <url>https://raw.githubusercontent.com/CodingCat/xgboost/maven-repo/</url>
+</repository>
+``` 
+
+and then refer to the snapshot dependency by adding:
+
+```xml
+<dependency>
+    <groupId>ml.dmlc</groupId>
+    <artifactId>xgboost4j</artifactId>
+    <version>next_version_num-SNAPSHOT</version>
+</dependency>
+```
+
+
+## Data Preparation
+
+As aforementioned, XGBoost4J-Spark seamlessly integrates Spark and XGBoost. The integration enables
+ users to apply various types of transformation over the training/test datasets with the convenient 
+ and powerful data processing framework, Spark.  
+ 
+In this section, we use [Iris](https://archive.ics.uci.edu/ml/datasets/iris) dataset as an example to
+ showcase how we use Spark to transform raw dataset and make it fit to the data interface of XGBoost.
+
+Iris dataset is shipped in CSV format. Each instance contains 4 features, "sepal length", "sepal width",
+"petal length" and "petal width". "class" column in each instance is essentially the label which has three possible values: "Iris Setosa", "Iris Versicolour" and "Iris Virginica". 
+
+### Read Dataset with Spark's Built-In Reader
+
+The first thing in data transformation is to load the dataset as Spark's structured data abstraction,
+DataFrame.
+
+```scala
+    import org.apache.spark.sql.SparkSession
+    import org.apache.spark.sql.types.{DoubleType, StringType, StructField, StructType}
+    
+    val spark = SparkSession.builder().getOrCreate()
+    val schema = new StructType(Array(
+      StructField("sepal length", DoubleType, true),
+      StructField("sepal width", DoubleType, true),
+      StructField("petal length", DoubleType, true),
+      StructField("petal width", DoubleType, true),
+      StructField("class", StringType, true)))
+    val rawInput = spark.read.schema(schema).csv("input_path")
+```
+
+At the first line, we create a instance of [SparkSession](http://spark.apache.org/docs/latest/sql-programming-guide.html#starting-point-sparksession) which is the entry of any Spark program working with DataFrame. The `schema` variable defines the schema of DataFrame wrapping Iris data. With this explicitly set schema, we can define the columns' name as well as their types, otherwise the column name would be the default ones derived by Spark, such as `_col0`, etc. Finally, we can use Spark's built-in csv reader to load Iris csv file as a DataFrame named `rawInput`.
+
+Spark also contains many built-in readers for other format. The latest version of Spark supports, csv/json/parquet/libsvm.
+ 
+### Transform Raw Iris Dataset
+
+To make Iris dataset be recognizable to XGBoost, we need to 
+
+1. Transform String-typed label, i.e. "class", to Integer-typed label.
+
+2. Assemble the feature columns as a vector to build XGBoost's internal data representation, DMatrix.
```

Comment:
Here I think the goal of assembling features into vector is to make it can fit into the ML pipeline, it's the same for all MLlib algorithms. I think we can mask the detail such as ```DMatrix```, as for XGBoost4J-Spark users, they never use ```DMatrix``` explicitly.